addons/account_edi/models/account_move.py

```diff
@@ -450,6 +453,36 @@ def _process_attachments_for_template_post(self, mail_template):
                 move_result.setdefault('attachments', []).extend(edi_attachments.get('attachments', []))
         return result
 
+    ####################################################
+    # Export Electronic Document
+    ####################################################
+
+    def action_download_electronic_invoice(self):
+        ids = [invoice.id for invoice in self if invoice.edi_document_ids]
+        if not ids:
+            return
+        url = '/web/edi_document/download?ids=' + ','.join(
+            str(id)
+            for id in ids
+        )
+
+        return {
+            'type': 'ir.actions.act_url',
+            'url': url,
+            'target': 'new',
+        }
+
+    def _create_zipped(self):
+        buffer = io.BytesIO()
+        zipfile_obj = zipfile.ZipFile(buffer, 'w')
+        for invoice in self:
+            for document in invoice.edi_document_ids:
+                if document.state in ['sent', 'cancelled']:
+                    name = (invoice.name + '_' + document.edi_format_name).replace('/', '_').replace(' ', '_').replace('.', '_').replace('(', '').replace(')', '') + '.xml'
```

Comment:
```suggestion
                    invoice_name = invoice.name.replace('/', '_')
                    edi_format_name = re.sub(r'[ ./]', '_', edi_format_name)
                    edi_format_name = re.sub(r'[()]', '', edi_format_name)
                    name = f'{invoice_name}_{edi_format_name}.xml'
```
What you did is correct, but I would prefer to split the logic as such
This is to avoid doing so many replace in a row, and also formatting the two names at once.
Finally this way we can use a string to make the final file name which is slightly better than concatenation